AwsCryptographicMaterialProviders/dafny/AwsCryptographyKeyStoreAdmin/test/TestMutationStateStructures.dfy

```diff
@@ -0,0 +1,144 @@
+  import AdminFixtures
+  import KMS = Com.Amazonaws.Kms
+  import DDB = Com.Amazonaws.Dynamodb
+
+  // Helper method to validate MutationDetails equality
+  method ValidateMutationDetails(actual: Types.MutationDetails, expected: Types.MutationDetails)
+  {
+    // Validate Original Properties
+    expect actual.Original.KmsArn == expected.Original.KmsArn;
+    expect actual.Original.EncryptionContext == expected.Original.EncryptionContext;
+    expect actual.Original.HierarchyVersion == expected.Original.HierarchyVersion;
+
+    // Validate Terminal Properties
+    expect actual.Terminal.KmsArn == expected.Terminal.KmsArn;
+    expect actual.Terminal.EncryptionContext == expected.Terminal.EncryptionContext;
+    expect actual.Terminal.HierarchyVersion == expected.Terminal.HierarchyVersion;
+
+    // Validate Mutations Input
+    expect actual.Input.TerminalKmsArn == expected.Input.TerminalKmsArn;
+    expect actual.Input.TerminalEncryptionContext == expected.Input.TerminalEncryptionContext;
+    expect actual.Input.TerminalHierarchyVersion == expected.Input.TerminalHierarchyVersion;
+
+    // Validate Remaining Commitment fields
+    expect actual.SystemKey == expected.SystemKey;
+    expect actual.CreateTime == expected.CreateTime;
+    expect actual.UUID == expected.UUID;
+  }
+
+  // Test a pre-HV-2 Mutation Commitment Deserializes Correctly
+  method {:test} TestPreHV2DeserializeMutation()
+  {
+    var underTest :- expect AdminFixtures.DefaultAdmin();
+
+    // Test case 1: System Key
+    var expectedDetails1 := Types.MutationDetails(
+      Original := Types.MutableBranchKeyContext(
+        KmsArn := "arn:aws:kms:us-west-2:370957321024:key/9d989aa2-2f9c-438c-a745-cc57d3ad0126",
+        EncryptionContext := map["aws-crypto-ec:ExampleContextKey" := "ExampleContextValue"],
+        HierarchyVersion := KeyStoreTypes.HierarchyVersion.v1
+      ),
+      Terminal := Types.MutableBranchKeyContext(
+        KmsArn := "arn:aws:kms:us-west-2:370957321024:key/bc127593-f7da-452c-a1f3-cd34c46f81f8",
+        EncryptionContext := map["aws-crypto-ec:Robbie" := "is a dog."],
+        HierarchyVersion := KeyStoreTypes.HierarchyVersion.v1
+      ),
+      Input := Types.Mutations(
+        TerminalKmsArn := Some("arn:aws:kms:us-west-2:370957321024:key/bc127593-f7da-452c-a1f3-cd34c46f81f8"),
+        TerminalEncryptionContext := Some(map["Robbie" := "is a dog."]),
+        TerminalHierarchyVersion := None
+      ),
+      SystemKey := "KMS Symmetric Encryption",
+      CreateTime := "2025-04-18T21:18:41.000871Z",
+      UUID := "0aa2b18c-c639-4558-bb30-158ac2a22571"
+    );
+
+    var response1 := underTest.DescribeMutation(Types.DescribeMutationInput(
+                                                  Identifier := AdminFixtures.STATIC_PRE_HV2_MUTATION_WITH_SYSTEM_KEY
+                                                ));
+    expect response1.Success?, "Mutation Commitment is present, however unable to deserialize.";
```

Comment:
Issue: I do not think the failure message is true, as the successful response may say `MutationInFlight.No`.
```suggestion
    expect response1.Success?;
```